File: codility/PassingCars.js
function solution(A) {
    let result = 0;
    let lastZeros = 0;
    for (let i = 0; i < A.length; i++) {
        const val = A[i];
        if (val === 0) {
            lastZeros++;
        } else {
            result += lastZeros;
            if (result > 1000000000) {
                result = -1;
                break;
            }
        }
    }
    return result;
}

File: codility/python/PassingCars.py
def solution(A):
    # Implement your solution here
    ones_before = 0
    zero_count = 0
    minus_count = 0
    for i in range(len(A)):
        if A[i] == 0:
            zero_count+=1
            minus_count+=ones_before
        else:
            ones_before+=1

    pairs = zero_count * (len(A) - zero_count) - minus_count
    return -1 if pairs > 1000000000 else pairs